sections/method/preprocessing.tex

```diff
@@ -0,0 +1,138 @@
+The two databases used are:  Wiktionary\footnote{https://github.com/fnielsen/awesome-danish (specifically Danish resources: https://people.compute.dtu.dk/faan/ps/Nielsen2016Danish.pdf)} and DanNet\cite{Pedersen2009DanNetTC}. 
+
+Most words remaining at this step are contained within the databases, so this step does not remove as many words, however, this is mostly due to the step being taken late, due to having a longer computation time per word.
+This step mostly serves to remove names, and words from different languages, namely English.
+However, it also cuts the odd non-words (e.g. 'gimi') and abbreviations (e.g. 'DK') that still exist at this point in the preprocessing.
+The example used in \autoref{prepro:example3} does not change at this step. 
+Instead, \autoref{prepro:example4} shows an example of another document that does change during this step.
+
+\begin{figure}[h]
+	\begin{framed}
+		\begin{quote}
+			\textit{
+				'\colorbox{pink}{country} kræver sommerens by \colorbox{pink}{night} sluttede afslappet band'
+			}
+		\end{quote}
+	\end{framed}
+	\caption{Snippet of a document which had words removed during the word database filtering. The highlighted words are removed, due to being foreign words.}
+	\label{prepro:example4}
+\end{figure}
+
+\paragraph{Lemmatization}
+Words are reduced to their lemmas using the python package \texttt{lemmy}\footnote{\url{https://pypi.org/project/lemmy/}}. 
+This is done by reducing word inflections.
+However since we do not have proper POS-tagging, some words have multiple meanings depending on the word class.
+In these cases we keep all inflected words.
+After lemmatization has been applied some words might have been converted into stopwords.
+We, therefore, remove stopwords once again.
+We also remove any potential documents that may have become empty after all the word filtering, however, this is usually not the case.
+\begin{figure}[h]
+	\begin{framed}
+		\begin{quote}
+			\textit{
+				'fingre dankort hobro atter årig adelgade dankort kode'
+			}
+		\end{quote}
+	\end{framed}
+	\caption{Lemmatized version of the example document from \autoref{prepro:example3}}
+	\label{prepro:example5}
+\end{figure}
+
+\paragraph{Construct Files}
+Finally, we construct and save the \texttt{CountVectorizer} (document-word frequency matrix) as a sparse matrix \texttt{npz} file.
+This file is used by the LDA to construct a document-topic matrix and a topic-word matrix.
+We also save various lookup files in \texttt{csv} format.
+\texttt{word2vec} and \texttt{doc2vec} files translate words and documents to their IDs.
+\texttt{doc2word} translates document IDs to all the IDs of the words used in the document.
+These files allow us to only use IDs in future phases, and then later translate back to words when needed.
+\todo[inline]{den skal nok bare flyttes til appendix}
```

Comment:
flyt til appendix eller slet

Reply:
hellere slet så, tror jeg. Tror ikke der er nogen brug for dette i appendix

Reply:
```suggestion
```